### PR TITLE
Regenerate Job Router with stream-style serialization and remove many Internal models

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Regenerates `azure-communication-jobrouter` with stream-style serialization enabled. Additionally, adds numerous code customizations so that the `*Internal` model mapping pattern can be reduced greatly, lowering maintenance costs and possible bugs when mapping handwritten models to generated models.

There will need to be a corresponding change in `azure-rest-api-specs` based on what I changed manually in `models.tsp` and `tspconfig.yml` while creating this PR. I will open that PR once this has approval.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@williamzhao87 could you take a look at this PR. It's onboarding Job Router to azure-json and using code customizations to remove a lot of the mapping logic from custom to generated types but having the generated types handle the same as the custom types.

**Review by mssfang — APPROVED** [review]

**Comment by williamzhao87** [comment]
@alzimmermsft Thanks for submitting the PR with these extensive changes! Everything here looks good to me.

**Review by williamzhao87 — APPROVED** [review]

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/models/CancelJobResultInternal.java [hunk 1]
```diff
@@ -5,17 +5,53 @@
...
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
 
 /**
  * Response payload from cancelling a job.
  */
 @Immutable
-public final class CancelJobResultInternal {
+public final class CancelJobResultInternal implements JsonSerializable<CancelJobResultInternal> {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/models/CancelJobResultInternal.java:18]: Just curious, seems like most of the internal classes are removed.  Is there something specific about these classes that make it not possible to remove?

Comment (alzimmermsft) [reply sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/models/CancelJobResultInternal.java:18]: I really never figured out what to do with this class and the similar `Internal` classes as they're defined in the TypeSpec without any properties but are used in service methods that are hidden from the public API, so I left them. These can be followed up on if they should be removed.

**Pull request merged**